src/mlpack/tests/random_forest_test.cpp

```diff
@@ -607,3 +607,117 @@ TEST_CASE("ExtraTreesAccuracyTest", "[RandomForestTest]")
 
   REQUIRE(accuracy >= 0.85);
 }
+
+/**
+ * Test ComputeAverageUniqueness.
+ */
+TEST_CASE("ComputeAverageUniquenessTest", "[RandomForestTest]")
+{
+  using SBS = SequentialBootstrap<true>;
+
+  arma::mat indM(3 /* rows */, 6 /* cols */, arma::fill::zeros);
+
+  indM(0, 0) = indM(0, 1) = indM(0, 2) = 1.0;
+  indM(1, 2) = indM(1, 3) = 1.0;
+  indM(2, 4) = indM(2, 5) = 1.0;
+
+  const arma::vec avg(SBS::ComputeAverageUniqueness(indM));
+
+  REQUIRE(avg(0) == 5.0 / 6.0);
+  REQUIRE(avg(1) == 0.75);
+  REQUIRE(avg(2) == 1.0);
+}
+
+/**
+ * Test ComputeNextDrawProbabilities.
+ */
+TEST_CASE("ComputeNextDrawProbabilitiesTest", "[RandomForestTest]")
+{
+  using SBS = SequentialBootstrap<true>;
+
+  const arma::uvec phi1(1, arma::fill::value(1));
+  arma::mat indM(3 /* rows */, 6 /* cols */, arma::fill::zeros);
+
+  indM(0, 0) = indM(0, 1) = indM(0, 2) = 1.0;
+  indM(1, 2) = indM(1, 3) = 1.0;
+  indM(2, 4) = indM(2, 5) = 1.0;
+
+  const arma::vec delta2(SBS::ComputeNextDrawProbabilities(phi1, indM));
+
+  REQUIRE(delta2(0) == 5.0 / 14.0);
+  REQUIRE(delta2(1) == 3.0 / 14.0);
+  REQUIRE(delta2(2) == 6.0 / 14.0);
+}
+
+struct Generator {
+  using result_type = unsigned;
+
+  constexpr static result_type min()
+  {
+    return 0;
+  }
+
+  constexpr static result_type max()
+  {
+    return 2;
+  }
+
+  result_type operator()() const
+  {
+    static result_type random(0);
+    result_type rand(random);
+
+    random = 1;
+    return rand;
+  }
+};
```

Comment:
Could you just use one of the already-existing [random functions](https://www.mlpack.org/doc/user/core/math.html#rng-and-random-number-utilities) here?  It seems like `Random(0, 2)` would work here.

Reply:
In this unit test I don't want to get a random number. I want to get a predictable number so that I can base the test on it.